connection.go

```diff
@@ -486,7 +489,28 @@ func (c *Connection) readLoop() {
 			break
 		}
 
-		go c.handleResponse(rawMessage)
+		c.readResponseCh <- rawMessage
+	}
+
+	c.handleConnectionError(err)
+}
+
+func (c *Connection) readResponseLoop() {
+	var err error
+
+	for err == nil {
+		for {
+			select {
+			case mess := <-c.readResponseCh:
+				go c.handleResponse(mess)
+			case <-time.After(c.Opts.ReadTimeout):
+				if c.Opts.ReadTimeoutHandler != nil {
+					go c.Opts.ReadTimeoutHandler(c)
+				}
```

Comment:
minor / optional: did we want to log or error or something if we hit this case and `ReadTimeoutHandler` is nil for some reason?

Reply:
Like the other handlers, `nil` just means we ignore it. It's fair to assume there could be implementations where long intervals between reads are not unusual.